lib/ui/semantics.dart

```diff
@@ -818,7 +818,7 @@ abstract class SemanticsUpdateBuilder {
     required String hint,
     required List<StringAttribute> hintAttributes,
     required String tooltip,
-    required TextDirection? textDirection,
+    required TextDirection textDirection,
```

Comment:
I am surprised this is not breaking the framework https://github.com/flutter/flutter/blob/28b81dc575da37247cb0719b17883b8db65560f6/packages/flutter/lib/src/semantics/semantics.dart#L2799
data.textDirection is nullable.

do you have another pr to migrate framework code?

Reply:
I am also surprised. I was suspecting that this PR will have red CI, and then I'd submit a separate framework PR.

Reply:
It looks like they are disabled here https://github.com/flutter/engine/pull/50800